### PR TITLE
[PORT] Makes rads glow, buffs rad goggles

### PR DESCRIPTION
**Pull request by Moccha-Bee**
## About The Pull Request
I've made contaminated objects pulse bright green, and increased the range and duration of rad goggles, along with tweaking the color and fixing an issue with the layer the text was drawn on. Oh and I upped the alpha to 255, there's a lot of info in one show of the numbers, it should be as easy to read as possible.

 Original author: I've added mobs to the contaminate list, from talking to @ninjanomnom the reason it was put on the the blacklist was because of golems going around immune to rads, I think this will partially help that, as it's very obvious where the rads are coming from, along with the fact that this strat is basically just worse plasmemes.

The glow varies in phase, so it shouldn't look..... static or mechanical. I don't thing varying duration is a good idea, but let me know what you think.

The code behind it is just applying an outline to the object, so far as I can tell the worst it'll look is demonstrated in the window in the below picture. I genuinely have no idea what causes that, but I don't think it's a major issue.

image

The change will also highlight what doesn't get irradiated, mostly walls, I'm not sure if that's an issue or not.

https://github.com/tgstation/tgstation/pull/51879

## Why It's Good For The Game
Mostly gonna argue for the glow change here, not so much the rad goggles. But in short most people thought they didn't work, and they were not that usable due to some settings.

Now, onto the glow thing.

First of all, all this is being done with filters and animate(), both of which are mostly client side. The addition shouldn't cause any massive lag on the servers end, even during massive rad waves. A clump might cause slight client side lag, but that could be down to me running the server and the client at the same time.

As for why it's a good idea:
Radiation is bad for 3 main reasons. The primary one being that it is laggy as hell as soon as it starts to contaminate things, which should be helped by #51693, secondly it's very hard to track or contain. A foolish atmos tech is like a virus, and it spreads fast. This should help with that immensely. Finally the radiation system is really a bit jank. Not contamination mind you, that behaves sanely if not fast, I mean the way it interacts with mobs and items. This is a general issue that will need to be addressed in future.

Oh and RE:Realism, this game is less about how things work and more about how people think things work, trimmed and tweaked to be fun. Glowing green rads is on point in that sense, at least in my opinion.

One more thing, thanks to the fruit for this idea like 3 months ago. I got bored and figured I'd do it.

## Changelog
:cl:
add: Contaminated objects will now glow bright green until cleaned, remember to quarantine rouge free golems.
tweak: Buffed rad goggles in general, the text lasts longer, is brighter, and will show more then twice as far away
fix: Fixed the text from radiation goggles not showing above mobs
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
the rad glowing effect looks ugly imo

**Comment by Moccha-Bee** [comment]
> the rad glowing effect looks ugly imo

image
I think the graphics look worse in my end, this one is the original image from the pr. which looks much better.

**Comment by Isy232** [comment]
@Moccha-Bee If you ever get the rads on an object into scientific notation using rad goggles will instantly crash your game.

Unless that was patched some time ago.

I think this is a great PR, but if you could deal with that if it is still a problem it would be even better.

**Comment by Moccha-Bee** [comment]
> @Moccha-Bee If you ever get the rads on an object into scientific notation using rad goggles will instantly crash your game.
> 
> Unless that was patched some time ago.
> 
> I think this is a great PR, but if you could deal with that if it is still a problem it would be even better.

I tried looking for a pr fixing that bug but couldn't find it.

File: code/__DEFINES/layers.dm [hunk 1]
```diff
@@ -88,6 +88,8 @@
 #define LIGHTING_PLANE 15
 #define LIGHTING_LAYER 15
 
+#define RAD_TEXT_LAYER 15.1
```
Comment (qwertyquerty) [inline code/__DEFINES/layers.dm]: Shouldn't it just use the hud layer

**Comment by qwertyquerty** [comment]
I think the rads would look better if it was a more spread and blurred outline

File: code/__DEFINES/layers.dm (hunk 1 above)
Comment (Moccha-Bee) [reply code/__DEFINES/layers.dm]: The creator decided to use a custom layer, the normal hud layer works just fine want me to make it use the hud layer?

Comment (qwertyquerty) [reply code/__DEFINES/layers.dm]: Yeah

**Comment by PowerfulBacon** [comment]
Personally, I think this looks really ugly and would be better if it was only shown while the goggles are worn (I know this is impossible). I would be fine with just the numbers and not the effects, but I know zesko does like this so we are kind of mixed overall

File: code/__DEFINES/layers.dm [hunk 2]
```diff
@@ -88,6 +88,8 @@
 #define LIGHTING_PLANE 15
 #define LIGHTING_LAYER 15
 
+#define RAD_TEXT_LAYER 21
```
Comment (qwertyquerty) [inline code/__DEFINES/layers.dm:91]: Why define this to be the same as the hud layer instead of just putting it on the hud layer?

**Comment by qwertyquerty** [comment]
I still think it needs to be a wider and more blurred glow

**Comment by Moccha-Bee** [comment]
> Personally, I think this looks really ugly and would be better if it was only shown while the goggles are worn (I know this is impossible). I would be fine with just the numbers and not the effects, but I know zesko does like this so we are kind of mixed overall

The rads only show with the goggles or CE hardsuit (dont know if youbwere being sarcastic) and like i said it looks pretty ugly on local server but this was due to some setting iirc.

**Review by ike709 — APPROVED** [review]
Seemed fine in testmerge.

**Comment by PowerfulBacon** [comment]
Better than I thought it would be

**Pull request merged**